File: JavaPolyominoes/src/io/github/ascenderx/polyominoes/views/GameWindow.java
package io.github.ascenderx.polyominoes.views;

import java.awt.event.KeyEvent;

import javax.swing.JFrame;

import io.github.ascenderx.polyominoes.models.InputHandler;

public class GameWindow extends JFrame {
  private InputHandler inputHandler;
  private static final int SCALE = 4;
  private static final int WIDTH = 128 * SCALE;
  private static final int HEIGHT = 64 * SCALE;
  private static final String TITLE = "Tiles";
  private static final long serialVersionUID = 1L;

  public GameWindow(GameCanvas canvas, InputHandler inputHandler) {
    super(TITLE);

    // Set properties.
    setSize(WIDTH, HEIGHT);
    setResizable(false);
    setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
    
    // Add components.
    add(canvas);

    // Register input handler.
    this.inputHandler = inputHandler;
  }

  public void reveal() {
    setVisible(true);
  }

  @Override
  protected void processKeyEvent(KeyEvent event) {
    // TODO: Bind to an adapter.
  }
}

File: Kotlin/KotlinPolyominoes/src/io/github/ascenderx/Polyominoes/views/GameWindow.kt
package io.github.ascenderx.polyominoes.views

import java.awt.event.KeyEvent

import javax.swing.JFrame

import io.github.ascenderx.polyominoes.models.InputHandler

class GameWindow(canvas: GameCanvas, inputHandler: InputHandler) : JFrame(TITLE) {
  var inputHandler: InputHandler = inputHandler
  
  companion object {
    val SCALE: Int = 4
    val WIDTH: Int = 128 * SCALE
    val HEIGHT: Int = 64 * SCALE
    val TITLE: String = "Tiles"
    val serialVersionUID: Long = 1L;
  }
  
  init {
    setSize(WIDTH, HEIGHT)
    setResizable(false)
    setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)
    
    add(canvas)
  }
  
  fun reveal() {
    setVisible(true)
  }
  
  override fun processKeyEvent(event: KeyEvent) {
    
  }
}
